src/main/java/seedu/tatoolkit/ui/MainWindow.java

```diff
@@ -187,8 +187,13 @@ private CommandResult executeCommand(String commandText) throws CommandException
             logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
 
-            Optional<Person> lastViewedPerson = logic.getLastViewedPerson();
-            lastViewedPerson.ifPresentOrElse(sidePanel::displayPerson, sidePanel::resetDetails);
+            logic.getLastExecutedCommand().ifPresent(command -> {
+                if (command instanceof ViewCommand) {
+                    logic.getLastViewedPerson().ifPresentOrElse(sidePanel::displayPerson, sidePanel::resetDetails);
+                } else if (command instanceof ListCommand) {
+                    sidePanel.displayAttendanceList(logic.getFilteredPersonAttendanceList());
```

Comment:
Just to clarify, this code checks the lastExecutedCommand to determine if the `displayAttendanceList` or `displayPerson` should be displayed on the side panel.

However, could this reintroduce the bug where if a person is currently displayed, and we do a edit command, then the displayed person will not be updated.

Instead, is it possible to have a variable to keep track of whether the side panel should is currently on `displayPerson` or `displayAttendanceList`, then execute the function every time to make sure it is always updated.

Reply:
> is it possible to have a variable to keep track of whether the side panel should is currently on displayPerson or displayAttendanceList
This is what the `lastExecutedCommand` variable does.
If a person is currently displayed on the side panel, then the `LastExecutedCommand` has to be ViewCommand, so it would update it dynamically every time `MainWindow.executeCommand()` is called.

Reply:
> Instead, is it possible to have a variable to keep track of whether the side panel should is currently on `displayPerson` or `displayAttendanceList`, then execute the function every time to make sure it is always updated.
> 
This is what the `lastExecutedCommand` variable does.
If a person is currently displayed on the side panel, then the `LastExecutedCommand` has to be ViewCommand, so it would update it dynamically every time `MainWindow.executeCommand()` is called.

Reply:
If I do a ViewCommand, then an EditCommand, then would the lastExecutedCommand be an EditCommand or a ViewCommand?

Reply:
ViewCommand. lastExecutedCommand is exclusive to either View or List command as of now.
In `LogicManager.execute()`, this is the logic:
```
if (command instanceof ViewCommand || command instanceof ListCommand) {
            lastExecutedCommand = Optional.of(command);
        }
```

Reply:
I admit, lastExecutedCommand might not be the most appropriate name. Any ideas?

Reply:
Nice! Maybe lastSidePanelDisplay?

Reply:
Okay, changed `lastExecutedCommand` to `lastSidePanelCommand`